MMOCoreORB/src/server/zone/objects/building/BuildingObjectImplementation.cpp

```diff
@@ -1193,10 +1194,8 @@ void BuildingObjectImplementation::payAccessFee(CreatureObject* player) {
 	if (owner != nullptr && owner->isPlayerCreature()) {
 		Locker clocker(owner, player);
 
-		PlayerObject* ghost = owner->getPlayerObject();
-
-		if (ghost != nullptr)
-			ghost->addExperience("merchant", 50, true);
+		PlayerManager* playerManager = getZoneServer()->getPlayerManager();
+		playerManager->awardExperience(player, "merchant", 50, false);
```

Comment:
Same here, no null check

Reply:
yeah she removed ghost so do not need nullptr check.

Reply:
Ah yeah this one is good